tfhe/c_api_tests/test_high_level_boolean.c

```diff
@@ -0,0 +1,102 @@
+#include <tfhe.h>
+
+#include <stdio.h>
+#include <inttypes.h>
+#include <assert.h>
+
+#define GOTO_IF_TRUE(label, cond) if ((cond) == 1) {\
+  goto label;\
+  }
+
+int client_key_test(const ClientKey *client_key) {
+    int ok;
+    FheBool *lhs = NULL;
+    FheBool *rhs = NULL;
+    FheBool *result = NULL;
+
+    bool lhs_clear = 0;
+    bool rhs_clear = 1;
+
+    ok = fhe_bool_try_encrypt_with_client_key_bool(lhs_clear, client_key, &lhs);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    ok = fhe_bool_try_encrypt_with_client_key_bool(rhs_clear, client_key, &rhs);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    ok = fhe_bool_bitand(lhs, rhs, &result);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    bool clear;
+    ok = fhe_bool_decrypt(result, client_key, &clear);
+    GOTO_IF_TRUE(exit, ok != 0)
+
+    assert(clear == (lhs_clear & rhs_clear));
+
+exit:
+    fhe_bool_destroy(lhs);
+    fhe_bool_destroy(rhs);
+    fhe_bool_destroy(result);
```

Comment:
I think we may want to add safe destroy to boolean and shortint the same way you do it here, I'll create an issue